detectors/resources/src/test/java/com/google/cloud/opentelemetry/detectors/GKEResourceTest.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.opentelemetry.detectors;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class GKEResourceTest {
+  @Rule public final WireMockRule wireMockRule = new WireMockRule(8089);
+  private final GCPMetadataConfig metadataConfig = new GCPMetadataConfig("http://localhost:8089/");
+
+  // Helper method to help stub endpoints
+  private void stubEndpoint(String endpointPath, String responseBody) {
+    stubFor(
+        get(urlEqualTo(endpointPath))
+            .willReturn(
+                aResponse().withHeader("Metadata-Flavor", "Google").withBody(responseBody)));
+  }
+
+  @Test
+  public void testGKEResourceNotGCP() {
+    GKEResource test = new GKEResource();
+
+    /* The default meta data url is unreachable through testing so getAttributes should not detect a
+    GCP environment, hence returning empty attributes */
```

Comment:
This is correct for "GKE on GCP" (most common) but there are actually GKE distros to run on other platforms (Anthos), e.g. [GKE on AWS](https://cloud.google.com/anthos/clusters/docs/aws/concepts/architecture) or [GKE on-premise](https://cloud.google.com/anthos/clusters/docs/on-prem/1.3/overview).

Still, this is probably fine to merge as is, and add support for GKE outside GCP later.